src/middlewares/userMiddlewares.js

```diff
@@ -47,4 +47,23 @@ export default class UserMiddlewares {
       return resError(res, 500, error.message);
     }
   }
+
+  /**
+ * @name confirmUserExists
+ * @description Checks if a user exists in the database
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @param {object} next The response object
+ * @returns {object} The API response or next()
+ */
+  static async confirmUserExists(req, res, next) {
```

Comment:
This function is a sort of a duplicate function, adding this would mean we now have three different middlewares that call the same user service. I've looked at the code, I think this may be okay for now, because trying to combine it may make you touch other aspects of the code that this PR doesn't cover. we may have to open a branch that refactors this.

Reply:
There is a `checkUserExistBeforeLogin` function in the codebase, reuse that, but rename it to `checkIfUserExist`, remember to also rename this in this login route where it was used

Reply:
The `checkifUserExistBeforeLogin` checks if user exist during login and gets user email from req.body.
However, the `confirmUserExists` checks if user is valid, during registration. It gets user email  from token sent to the user. As user should not be allowed to update his email during registration.

The two middleware are not exactly the same. Please advise.

Reply:
Okay I have seen this, I would say that you redactor but that might still delay this PR so I will create a chore on the PT board for it instead.

Reply:
Okay